demos/solps_from_balance.py

```diff
@@ -0,0 +1,163 @@
+xrange = np.linspace(xl, xu, 500)
+yrange = np.linspace(yl, yu, 500)
+
+
+for i, x in enumerate(xrange):
+    for j, y in enumerate(yrange):
+        ne_samples[j, i] = plasma.electron_distribution.density(x, 0.0, y)
+        te_samples[j, i] = plasma.electron_distribution.effective_temperature(x, 0.0, y)
+        d0_samples[j, i] = d0.distribution.density(x, 0.0, y)
+        d1_samples[j, i] = d1.distribution.density(x, 0.0, y)
+        c0_samples[j, i] = c0.distribution.density(x, 0.0, y)
+        c1_samples[j, i] = c1.distribution.density(x, 0.0, y)
+        c2_samples[j, i] = c2.distribution.density(x, 0.0, y)
+        c3_samples[j, i] = c3.distribution.density(x, 0.0, y)
+        c4_samples[j, i] = c4.distribution.density(x, 0.0, y)
+        c5_samples[j, i] = c5.distribution.density(x, 0.0, y)
+        c6_samples[j, i] = c6.distribution.density(x, 0.0, y)
+
+mesh.plot_mesh()
+plt.title('mesh geometry')
+
+plt.figure()
+plt.imshow(ne_samples, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("electron density")
+
+plt.figure()
+plt.imshow(te_samples, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("electron temperature")
+
+plt.figure()
+plt.imshow(d0_samples, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("D0 density")
+plt.figure()
+plt.imshow(d1_samples, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("DI density")
+
+plt.figure()
+plt.imshow(c0_samples, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("CI density")
+plt.figure()
+plt.imshow(c1_samples, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("CII density")
+plt.figure()
+plt.imshow(c2_samples, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("CIII density")
+plt.figure()
+plt.imshow(c3_samples, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("CIV density")
+plt.figure()
+plt.imshow(c4_samples, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("CV density")
+plt.figure()
+plt.imshow(c5_samples, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("CVI density")
+plt.figure()
+plt.imshow(c6_samples, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("CVII density")
+
+plt.figure()
+plt.imshow(d0_velocity, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("D0 velocity")
+
+plt.figure()
+plt.imshow(d1_velocity, extent=[xl, xu, yl, yu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(yl, yu)
+plt.title("D1 velocity")
```

Comment:
The D0 and D1 bulk velocities are not defined, so these plots will fail. Delete this section.